### PR TITLE
hikey960: Add initial support

### PR DESCRIPTION
**Pull request by vchong**
Signed-off-by: Victor Chong <victor.chong@linaro.org>

### CODE REVIEW

**Comment by vchong** [comment]
Depends on https://github.com/OP-TEE/manifest/pull/67.

File: hikey960.mk [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+
+.PHONY: linux-cleaner
+linux-cleaner: linux-cleaner-common
+
+################################################################################
+# OP-TEE
+################################################################################
+OPTEE_OS_COMMON_FLAGS += PLATFORM=hikey-hikey960 \
+			CFG_CONSOLE_UART=$(CFG_CONSOLE_UART) \
+			CFG_SECURE_DATA_PATH=n
```
Comment (jforissier) [inline hikey960.mk]: Why `CFG_SECURE_DATA_PATH=n`?

File: hikey960.mk [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+	rm -f $(STRACE_PATH)/Makefile $(STRACE_PATH)/configure
+
+################################################################################
+# Root FS
+################################################################################
+# Read stdin, expand ${VAR} environment variables, output to stdout
+# http://superuser.com/a/302847
+define expand-env-var
+awk '{while(match($$0,"[$$]{[^}]*}")) {var=substr($$0,RSTART+2,RLENGTH -3);gsub("[$$]{"var"}",ENVIRON[var])}}1'
+endef
```
Comment (jforissier) [inline hikey960.mk]: Duplicated from `hikey.mk`. Should be moved to a common file, such as `common.mk`.

File: hikey960.mk [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+EDK2_ARCH ?= AARCH64
+EDK2_DSC ?= OpenPlatformPkg/Platforms/Hisilicon/HiKey960/HiKey960.dsc
+EDK2_TOOLCHAIN ?= GCC5
+
+ifeq ($(CFG_CONSOLE_UART),5)
+	EDK2_BUILDFLAGS += -DSERIAL_BASE=0xFDF05000
+endif
+
+define edk2-call
+	$(EDK2_TOOLCHAIN)_$(EDK2_ARCH)_PREFIX=$(GCC5_AARCH64_CROSS_COMPILE) \
```
Comment (jforissier) [inline hikey960.mk]: Can we avoid requiring yet another toolchain? I.e., either build with $(LEGACY_AARCH64_CROSS_COMPILE) or $(AARCH64_CROSS_COMPILE).

File: hikey960.mk [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+# Following variables defines how the NS_USER (Non Secure User - Client
+# Application), NS_KERNEL (Non Secure Kernel), S_KERNEL (Secure Kernel) and
+# S_USER (Secure User - TA) are compiled
+################################################################################
+COMPILE_NS_USER   ?= 64
+override COMPILE_NS_KERNEL := 64
+COMPILE_S_USER    ?= 32
+COMPILE_S_KERNEL  ?= 64
+
+# Normal and secure world console UART: 6 (v2 or newer board) or 5 (v1 board)
```
Comment (jforissier) [inline hikey960.mk]: Doh! Did they really manage to produce incompatible versions of the board already? Anyway...

Comment (vchong) [inline hikey960.mk]: I think v1 boards are first run engineering boards only and probably not owned by many.

File: hikey960.mk (hunk 3 above)
Comment (vchong) [inline hikey960.mk]: Agree. My plan is to transition hikey.mk and hikey_debian.mk to this toolchain as well and remove legacy, but fvp and qemu v8 still use it. @jenswi-linaro Do you know if both these can also transition to gcc5? rpi3 also use it for u-boot. @igoropaniuk Same question?

File: hikey960.mk (hunk 1 above)
Comment (vchong) [inline hikey960.mk]: The kernel has some ion hacks that made it incompatible with our ion patches.

File: hikey960.mk (hunk 2 above)
Comment (vchong) [inline hikey960.mk]: Moved.

File: hikey960.mk (hunk 3 above)
Comment (vchong) [inline hikey960.mk]: @jforissier I've reverted back to LEGACY_AARCH64_CROSS_COMPILE for now. Upgrade to gcc5 or higher should be done later. Please continue review. Thanks!

**Comment by jforissier** [comment]
Thanks Victor.
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by vchong** [comment]
Rebased, squashed and tag added. Thanks!

**Pull request merged**